### PR TITLE
Issue #SC-776 feat:providing orgName in user search api response

### PR DESCRIPTION
**Pull request by amit-tarento**

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 1]
```diff
@@ -92,6 +101,62 @@ public void onReceive(Request request) throws Throwable {
     }
   }
 
+  @SuppressWarnings("unchecked")
+  private void fetchQueryParamDetails(String requestFields, Map<String, Object> userMap) {
+
+    if (StringUtils.isNotBlank(requestFields)) {
+      try {
+        if (requestFields.equalsIgnoreCase(JsonKey.ORG_NAME)) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: expect list of fields

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 2]
```diff
@@ -92,6 +101,62 @@ public void onReceive(Request request) throws Throwable {
...
+          list.add(JsonKey.ID);
+          list.add(JsonKey.ORG_NAME);
+
+          searchDTO.setFields(list);
+
+          Map<String, Object> filters = new HashMap<>();
+          filters.put(JsonKey.ID, orgIds);
+
+          searchDTO.getAdditionalProperties().put(JsonKey.FILTERS, filters);
+          List<Organisation> orgList = orgClient.esSearchOrgBySearchDto(searchDTO);
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: esSearchOrgbyOrgIds(List orgIds, List outputColumns)

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 3]
```diff
@@ -92,6 +101,62 @@ public void onReceive(Request request) throws Throwable {
...
+
+          userOrgList.forEach(
+              userOrg -> {
+                String userOrgId = (String) userOrg.get(JsonKey.ORGANISATION_ID);
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: userOrgId -> orgId

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 4]
```diff
@@ -92,6 +101,56 @@ public void onReceive(Request request) throws Throwable {
     }
   }
 
+  @SuppressWarnings("unchecked")
+  private void fetchQueryParamDetails(List<String> requestFields, Map<String, Object> userMap) {
+
+    if (CollectionUtils.isNotEmpty(requestFields)) {
+      try {
+        if (requestFields.contains(JsonKey.ORG_NAME)) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: contains check need to be case-insensitive

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 5]
```diff
@@ -92,6 +102,88 @@ public void onReceive(Request request) throws Throwable {
...
+        .stream()
+        .forEach(
+            userMap -> {
+              String rootOrgId = (String) userMap.get(JsonKey.ROOT_ORG_ID);
+              if (StringUtils.isNotBlank(rootOrgId)) {
+                orgIdList.add(rootOrgId);
+              }
+              List<Map<String, Object>> userOrgList =
+                  (List<Map<String, Object>>) userMap.get(JsonKey.ORGANISATIONS);
+              userOrgList
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: Add null check.

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 6]
```diff
@@ -92,6 +102,88 @@ public void onReceive(Request request) throws Throwable {
...
+                      userOrg -> {
+                        String userOrgId = (String) userOrg.get(JsonKey.ORGANISATION_ID);
+                        if (StringUtils.isNotBlank(userOrgId)) {
+                          orgIdList.add(userOrgId);
+                        }
+                      });
+            });
+    List<String> outputColumns = new ArrayList<>();
+    outputColumns.add(JsonKey.ID);
+    outputColumns.add(JsonKey.ORG_NAME);
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: Add supported fields from the requested fields only.

File: actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java [hunk 7]
```diff
@@ -92,6 +102,88 @@ public void onReceive(Request request) throws Throwable {
...
+                    }
+                    List<Map<String, Object>> userOrgList =
+                        (List<Map<String, Object>>) userMap.get(JsonKey.ORGANISATIONS);
+                    userOrgList
+                        .stream()
+                        .forEach(
+                            userOrg -> {
+                              String userOrgId = (String) userOrg.get(JsonKey.ORGANISATION_ID);
+                              if (StringUtils.isNotBlank(userOrgId)) {
+                                Organisation org = orgMap.get(rootOrgId);
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/search/SearchHandlerActor.java]: userOrgId

**Pull request merged**